plugins/ecue.js

```diff
@@ -116,123 +116,157 @@ function exportHandleModes(fixture, defaults, physical, xmlMan) {
     let viewPosCount = 1;
     for (let dmxCount=0; dmxCount<mode.channels.length; dmxCount++) {
       let chKey = mode.channels[dmxCount];
+      if ('switchingChannels' in fixture && chKey in fixture.switchingChannels) {
+        const switchingChannel = fixture.switchingChannels[chKey];
+        const defaultValue = fixture.availableChannels[switchingChannel.triggerChannel].defaultValue;
 
-      if (chKey === null) {
-        // we already handled this as part of a 16-bit channel or it is a undefined channel, so just skip
-        continue;
+        fixture.availableChannels[chKey] = Object.assign(
+          {}, fixture.availableChannels[getDefaultSwitchedChKey(defaultValue, switchingChannel)]
+        );
       }
 
-      let doubleByte = false;
-      const multiByteChannels = getCorrespondingMultiByteChannels(chKey, fixture);
-      if (multiByteChannels !== null
+      viewPosCount = parseChannel(chKey, mode, dmxCount, viewPosCount, xmlFixture);
+    }
+  }
+
+  function getDefaultSwitchedChKey(defaultValue, switchingChannel) {
+    for (const switchedChannelKey in switchingChannel.channels) {
+      const ranges = switchingChannel.channels[switchedChannelKey];
+
+      if (!Array.isArray(ranges[0])) {
+        if (ranges[0] <= defaultValue && defaultValue <= ranges[1]) {
```

Comment:
you could do `ranges = [ranges];` here and just fall through into the other case to reduce code duplication.